x-pack/plugins/security_solution/public/timelines/components/timeline/body/data_driven_columns/index.tsx

```diff
@@ -441,9 +441,16 @@ export const getMappedNonEcsValue = ({
   data,
   fieldName,
 }: {
-  data: TimelineNonEcsData[];
+  data?: TimelineNonEcsData[];
   fieldName: string;
 }): string[] | undefined => {
+  /*
+   While data _should_ always be defined
+   There is the potential for race conditions where a component using this function
+   is still visible in the UI, while the data has since been removed.
+   To cover all scenarios where this happens we'll check for the presence of data here
+  */
+  if (!data || data.length === 0) return undefined;
```

Comment:
Actually best to leave this as it more accurately represents a state that can exist for this component where data is not provided.